src/main/java/jewellery/inventory/mapper/ResourceMapper.java

```diff
@@ -43,4 +45,25 @@ public Resource toResourceEntity(ResourceRequestDto resourceRequestDto) {
     }
     throw new MappingException(resourceRequestDto);
   }
+
+  public String getGemstoneSizeString(Gemstone gemstone) {
+    return String.format("%.1fx%.1fx%.1f", gemstone.getDimensionX(), gemstone.getDimensionY(), gemstone.getDimensionZ());
+  }
+  public void setDimensions(GemstoneRequestDto dto) {
+    String size = dto.getSize();
+    if (size == null) {
+      throw new IllegalArgumentException("Size cannot be null");
+    }
+    String[] dimensions = size.split("x");
+    if (dimensions.length != 3) {
+      throw new IllegalArgumentException("Invalid size format. Must be in the format '1.00x2.00x3.00'");
+    }
+    try {
+      dto.setDimensionX(Double.parseDouble(dimensions[0].trim()));
+      dto.setDimensionY(Double.parseDouble(dimensions[1].trim()));
+      dto.setDimensionZ(Double.parseDouble(dimensions[2].trim()));
+    } catch (NumberFormatException e) {
+      throw new IllegalArgumentException("Invalid size format. Must be in the format '1.00x2.00x3.00'");
+    }
+  }
```

Comment:
This method shouldn't be needed. The request is going to be the same, so dimensions will be part of it.